File: src/2025-02-27-deepclaude37-rust/javascript/exercises/practice/pig-latin/pig-latin.js
//
// This is only a SKELETON file for the 'Pig Latin' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const translate = (text) => {
  return text.split(' ').map(word => {
    const lowerWord = word.toLowerCase();
    let match;

    // Rule 1: Check for words starting with vowels, "xr", or "yt"
    if (/^(?:[aeiou]|xr|yt)/.test(lowerWord)) {
      return lowerWord + 'ay';
    }

    // Rule 3: Check for consonants followed by "qu"
    if ((match = /^([^aeiou]*qu)(.*)/.exec(lowerWord))) {
      return match[2] + match[1] + 'ay';
    }

    // Rule 4: Check for consonants followed by "y"
    if ((match = /^([^aeiou]+)(y.*)/.exec(lowerWord))) {
      return match[2] + match[1] + 'ay';
    }

    // Rule 2: Move consonants to end
    if ((match = /^([^aeiou]+)(.*)/.exec(lowerWord))) {
      return match[2] + match[1] + 'ay';
    }

    return lowerWord + 'ay';
  }).join(' ');
};

File: src/python-2025-02-24-deepclaude-rust/python/exercises/practice/pig-latin/pig_latin.py
import re

def translate(text):
    translated_words = []
    for word in text.split():
        if re.match(r'^([aeiou]|xr|yt)', word):
            translated = word + 'ay'
        elif re.match(r'^[^aeiou]*qu', word):
            translated = re.sub(r'^([^aeiou]*qu)(.*)', r'\2\1ay', word)
        elif re.match(r'^[^aeiou]+y', word):
            translated = re.sub(r'^([^aeiou]+)y(.*)', r'y\2\1ay', word)
        else:
            translated = re.sub(r'^([^aeiou]+)(.*)', r'\2\1ay', word)
        translated_words.append(translated)
    return ' '.join(translated_words)
